### PR TITLE
Refactor user states

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

**Comment by bgeuken** [comment]
This changes the API in the sense that it does no longer accept setting the states via the 'enum number'. 
We could also add some code that workarounds this specific case, but I would prefer not to.
And it the API already uses the 'stringified' state. So I think it'S fair to say that this was a hidden feature.

`
<person>
  <login>Admin</login>
  <email>root@localhost</email>
  <realname>OBS Instance Superuser</realname>
  <state>confirmed</state>
  <globalrole>Admin</globalrole>
  <watchlist/>
</person>
`

File: src/api/app/models/user.rb [hunk 1]
```diff
@@ -22,6 +22,9 @@ def groups(user)
 class User < ApplicationRecord
   include CanRenderModel
 
+  # Keep in sync with states defined in db/structure.sql
```
Comment (hennevogel) [inline src/api/app/models/user.rb]: How about also making this an [enum](http://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html) in the model?

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -73,7 +73,7 @@ Metrics/AbcSize:
 # Offense count: 416
 # Configuration parameters: CountComments, ExcludedMethods.
 Metrics/BlockLength:
-  Max: 697
+  Max: 698
```
Comment (hennevogel) [inline .rubocop_todo.yml]: tz tz tz 😜

File: src/api/app/views/webui/user/_dropdown_menu.html.haml [hunk 3]
```diff
@@ -16,7 +16,7 @@
     %li
       = sprite_tag('cancel')
       = link_to('Delete user', {controller: "user", action: "delete", user: user.login }, |
-                method: :delete, data: { confirm: "Are you sure?" }) |
+                method: :post, data: { confirm: "Are you sure?" }) |
```
Comment (hennevogel) [inline src/api/app/views/webui/user/_dropdown_menu.html.haml]: :thought_balloon:  why?

File: src/api/app/models/user.rb (hunk 1 above)
Comment (bgeuken) [reply src/api/app/models/user.rb]: @hennevogel Been there, done [that](https://github.com/openSUSE/open-build-service/pull/3211). I am okay to give it another try. But not in this PR, and not right now :sweat:

Comment (hennevogel) [reply src/api/app/models/user.rb]: But all the things you do then magically work! ;-)

File: src/api/app/views/webui/user/_dropdown_menu.html.haml (hunk 3 above)
Comment (bgeuken) [reply src/api/app/views/webui/user/_dropdown_menu.html.haml]: Looks like I messed up a rebase. I was refactoring this part in #4631

File: .rubocop_todo.yml (hunk 2 above)
Comment (bgeuken) [reply .rubocop_todo.yml]: Hey, I can't refactor all of OBS at once :laughing:

File: src/api/app/models/user.rb (hunk 1 above)
Comment (bgeuken) [reply src/api/app/models/user.rb]: Yes, but there were so many from what I remember. So I would prefer to not block this PR for that.

File: src/api/app/views/webui/user/_dropdown_menu.html.haml (hunk 3 above)
Comment (bgeuken) [reply src/api/app/views/webui/user/_dropdown_menu.html.haml]: @hennevogel updated

**Comment by bgeuken** [comment]
[P4: (5) Allow local users in LDAP mode](https://trello.com/c/tj5UVn89/694-p4-5-allow-local-users-in-ldap-mode)

**Comment by hennevogel** [comment]
ach codecov...

**Pull request merged**